flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -462,6 +463,23 @@ class TableImpl(
     wrap(operationTreeBuilder.dropColumns(fields, operationTree))
   }
 
+  override def map(mapFunction: String): Table = {
+    map(ExpressionParser.parseExpression(mapFunction))
+  }
+
+  override def map(mapFunction: Expression): Table = {
+    val resolvedMapFunction = callResolver.visit(mapFunction)
```

Comment:
How about using `mapFunction.accept(callResolver)`, just to use the callResolver in the TableImpl class is very uniform, and I am fine if you keep using `callResolver.visit(mapFunction)` in this PR. :)